parsers/JP_KN.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+import logging
+import datetime
+
+
+from PIL import Image
+from pytesseract import image_to_string
+from bs4 import BeautifulSoup
+from urllib.request import urlopen, Request
+from io import BytesIO
+import re
+
+# The arrow library is used to handle datetimes
+import arrow
+# The request library is used to fetch content through HTTP
+import requests
+from .JP import fetch_production as JP_fetch_production
+
+# please try to write PEP8 compliant code (use a linter). One of PEP8's
+# requirement is to limit your line length to 79 characters.
+
+
+def fetch_production(zone_key='JP-KN', session=None,
+                     target_datetime: datetime.datetime = None,
+                     logger: logging.Logger = logging.getLogger(__name__)):
+
+    """
+    This method adds nuclear production on top of the solar data returned
+    by the JP parser. It tries to match the solar data with the nuclear 
+    data. If there is a difference of more than 30 minutes between solar 
+    and nuclear data, the method will fail.
+    """
+    r = session or requests.session()
+    if target_datetime is not None:
+         raise NotImplementedError('This parser can only fetch live data')
+    
+    JP_data = JP_fetch_production(zone_key, session, target_datetime, logger)
+    nuclear_mw, nuclear_datetime = get_nuclear_production()
+    latest = JP_data[-1] #latest solar data is the most likely to fit with nuclear production
+    diff = None
+    if nuclear_datetime > latest["datetime"]:
+        diff = nuclear_datetime-latest["datetime"]
+    else:
+        diff = latest["datetime"]-nuclear_datetime
+    if abs(diff.seconds) > 30 * 60:
+        raise Exception("Difference between nuclear datetime and solar data is too large")
```

Comment:
A nitpick, but it's the difference between nuclear data and the rest of the data right (solar + unknown)? Maybe change the comments to be "JP data" or "production data".